pkg/status/conditions_test.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/operator-framework/operator-sdk/pkg/status"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+var (
+	initTime      time.Time
+	clockInterval time.Duration
+)
+
+func init() {
+	loc, _ := time.LoadLocation("Local")
+	initTime = time.Date(2015, time.July, 11, 0, 1, 0, 0, loc)
+	clockInterval = time.Hour
+}
+
+func initConditions(init ...status.Condition) status.Conditions {
+	// Use the same initial time for all initial conditions
+	status.Clock = clock.NewFakeClock(initTime)
+	conditions := status.Conditions{}
+	for _, c := range init {
+		conditions.SetCondition(c)
+	}
+
+	// Use an incrementing clock for the rest of the test
+	status.Clock = &clock.IntervalClock{
+		Time:     initTime,
+		Duration: clockInterval,
+	}
+
+	return conditions
+}
+
+func generateCondition(t status.ConditionType, s corev1.ConditionStatus) status.Condition {
+	c := status.Condition{
+		Type:    t,
+		Status:  s,
+		Reason:  status.ConditionReason(fmt.Sprintf("My%s%s", t, s)),
+		Message: fmt.Sprintf("Condition %s is %s", t, s),
+	}
+	return c
+}
+
+func withLastTransitionTime(c status.Condition, t time.Time) status.Condition {
+	c.LastTransitionTime = metav1.Time{Time: t}
+	return c
+}
+
+func TestConditionsSetEmpty(t *testing.T) {
+	conditions := initConditions()
+
+	setCondition := generateCondition("A", corev1.ConditionTrue)
+	assert.True(t, conditions.SetCondition(setCondition))
+
+	expectedCondition := withLastTransitionTime(setCondition, initTime.Add(clockInterval))
+	actualCondition := conditions.GetCondition(setCondition.Type)
+	assert.Equal(t, 1, len(conditions))
```

Comment:
Ideally, in go we should use its default testing lib and not asserts.
I mean do the tests for example like [here](https://github.com/dev4devs-com/postgresql-operator/blob/master/pkg/controller/backup/controller_test.go#L34)